raphael-solver/src/quality_upper_bound_solver/state.rs

```diff
@@ -17,27 +17,35 @@ impl ReducedState {
     pub fn from_simulation_state(
         mut state: SimulationState,
         settings: &SolverSettings,
-        durability_cost: i16,
+        durability_cost: u16,
     ) -> Self {
-        state.cp += state.effects.manipulation() as i16
-            * (Manipulation::base_cp_cost(&state, &settings.simulator_settings) / 8);
+        let mut refunded_durability = state.durability / 5 + 1;
+        // Assume Manipulation effect can be used to its full potential
+        refunded_durability += u16::from(state.effects.manipulation());
```

Comment:
Change in visited nodes is caused by this change. Before, Manipulation ticks were refunded based on the cost of using Manipulation. Now, Manipulation ticks are refunded based on the durability that they restore. The difference is that the durability cost may be lower than `Manipulation::CP_COST / 8`, so states with Manipulation effect now have a stricter quality upper bound, which causes more states to be pruned.